pkg/tsdb/azuremonitor/credentials.go

```diff
@@ -44,16 +44,17 @@ func getAuthType(cfg *setting.Cfg, jsonData *simplejson.Json) string {
 
 func getDefaultAzureCloud(cfg *setting.Cfg) (string, error) {
 	// Allow only known cloud names
-	cloudName := cfg.Azure.Cloud
+	cloudName := ""
+	if cfg != nil && cfg.Azure != nil {
```

Comment:
`getDefaultAzureCloud` is a private method `cfg` is never `nil`. so this check isn't required.

Reply:
it's because of a unit test (that was not properly setting the config)